### PR TITLE
tests: x/gamm pool JoinSwapExactAmountIn consistency tests

### PR DESCRIPTION
**Pull request by alexanderbez**
## Description

- Adds tests for `JoinSwapExactAmountIn` testing consistency.
- A wee bit of linting/cleanup

Closes: #1116

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: x/gamm/keeper/pool_service.go [hunk 1]
```diff
@@ -249,18 +249,22 @@ func (k Keeper) JoinSwapExactAmountIn(
 	}
 
 	sharesOut, err := pool.JoinPool(ctx, tokensIn, pool.GetSwapFee(ctx))
-	if err != nil {
+	switch {
```
Comment (alexanderbez) [inline x/gamm/keeper/pool_service.go:252]: note to reviewer: Converted cascading if/else to a switch, which is much much easier to grok

File: x/gamm/keeper/pool_service.go [hunk 2]
```diff
@@ -249,18 +249,22 @@ func (k Keeper) JoinSwapExactAmountIn(
...
+
+	case sharesOut.LT(shareOutMinAmount):
+		return sdk.ZeroInt(), sdkerrors.Wrapf(
+			types.ErrLimitMinAmount,
+			"too much slippage; needed a minimum of %s shares to pass, got %s",
+			shareOutMinAmount, sharesOut,
+		)
+
+	case sharesOut.LTE(sdk.ZeroInt()):
 		return sdk.Int{}, sdkerrors.Wrapf(types.ErrInvalidMathApprox, "share amount is zero or negative")
```
Comment (alexanderbez) [inline x/gamm/keeper/pool_service.go:264]: question to maintainers: why do we return `sdk.Int{}` in this error case, but `sdk.ZeroInt()` in others?

Comment (ValarDragon) [reply x/gamm/keeper/pool_service.go:265]: Its inconsistent, we should change it all to be one of the two

**Review by ValarDragon — APPROVED** [review]
LGTM!

**Pull request merged**